dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/jdbc/JdbcCustomIconStore.java

```diff
@@ -77,7 +77,12 @@ public CustomIcon getIconByKey(String key) {
   }
 
   @Override
-  public List<CustomIcon> getIconsByKeywords(String[] keywords) {
+  public List<CustomIcon> getIconsByKeywords(List<String> keywords) {
+
+    if (keywords.isEmpty()) {
```

Comment:
I would rather completely remove the `getAllIcons()` method and add `where keywords @> string_to_array(?,',')` to the sql only if keywords is not empty.

This way if something changes in the query you do not have 2 places to change

Reply:
done